src/prototype_creep_resources.js

```diff
@@ -1,9 +1,20 @@
 'use strict';
 
 Creep.prototype.harvesterBeforeStorage = function() {
-  var methods = [
-    Creep.getEnergy
-  ];
+  let methods = [];
+
+  if (this.memory.hasEnergy === undefined) {
+    this.memory.hasEnergy = (this.carry.energy == this.carryCapacity);
+  } else if (this.memory.hasEnergy && this.carry.energy === 0) {
+    this.memory.hasEnergy = false;
+  } else if (!this.memory.hasEnergy &&
+      this.carry.energy == this.carryCapacity) {
+    this.memory.hasEnergy = true;
+  }
```

Comment:
I would put this logic to 'getEnergy'. The task are constructed the way, that each task decides if it gets executed and nothing else is done.

So if you move this logic to the beginning of `getEnergy` and `return false`. It is the same as if you wouldn't add it.